### PR TITLE
feat: (iso)miR intersection

### LINKED ISSUE #68: Create BAM file of the collapsed SAM file
From the SAM file obtained in issue #67  create the BAM file and the corresponding BAI file.

### PR DESCRIPTION
**Pull request by deliaBlue**
Closes #68  and #69.

In order to perform the (iso)miR intersection the following rules had been added to the workflow:
- Turn extended miRNA annotations file into a `BED` one
- Create and sort a `BAM` file for the `SAM` file inculding the alignments that intersected with the extended premiR annotations
- Create `BAI` file
- Intersect the created `BAM` file with the annotations on the `BED` file

### CODE REVIEW

**Comment by deliaBlue** [comment]
The workflow runs correctly locally. Here, the problem seems to be related with  the `singularity`. I looked but _by the moment_ could not find the reason of it nor how to fix it.

Here's one chunk of the log error:  

```
[Fri May 19 20:28:21 2023]
rule trim_genome_seq_id:
    input: test_files/genome.fa.gz, ../scripts/trim_id_fasta.sh
    output: results/genome.processed.fa
    log: logs/local/genome_process.log
    jobid: 4
    reason: Missing output files: results/genome.processed.fa
    resources: tmpdir=/tmp

(zcat test_files/genome.fa.gz | ../scripts/trim_id_fasta.sh > results/genome.processed.fa) &> logs/local/genome_process.log
ERROR  : Session directory does not exist: /usr/share/miniconda/envs/mirflowz/var/singularity/mnt/session
ABORT  : Retval = 255
 singularity  exec --home /home/runner/work/mirflowz/mirflowz/test --bind /home/runner/work/mirflowz/mirflowz/test/../ /home/runner/work/mirflowz/mirflowz/test/.snakemake/singularity/d66d590f45d199c9eb7f1c3bdfd1dae8.simg bash -c 'set -euo pipefail;  (zcat test_files/genome.fa.gz | ../scripts/trim_id_fasta.sh > results/genome.processed.fa) &> logs/local/genome_process.log'
ERROR  : Session directory does not exist: /usr/share/miniconda/envs/mirflowz/var/singularity/mnt/session
ABORT  : Retval = 255
Full Traceback (most recent call last):
  File "/usr/share/miniconda/envs/mirflowz/lib/python3.9/site-packages/snakemake/executors/__init__.py", line 2577, in run_wrapper
    run(
  File "/home/runner/work/mirflowz/mirflowz/workflow/rules/prepare.smk", line 118, in __rule_trim_genome_seq_id
    input:
  File "/usr/share/miniconda/envs/mirflowz/lib/python3.9/site-packages/snakemake/shell.py", line 300, in __new__
    raise sp.CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command ' singularity  exec --home /home/runner/work/mirflowz/mirflowz/test --bind /home/runner/work/mirflowz/mirflowz/test/../ /home/runner/work/mirflowz/mirflowz/test/.snakemake/singularity/d66d590f45d199c9eb7f1c3bdfd1dae8.simg bash -c 'set -euo pipefail;  (zcat test_files/genome.fa.gz | ../scripts/trim_id_fasta.sh > results/genome.processed.fa) &> logs/local/genome_process.log'' returned non-zero exit status 255.
```

**Review by uniqueg — CHANGES_REQUESTED** [review]
Points to address were discussed elsewhere. Briefly:

- No need to convert GFF to BED as GFF can be used with `bedtools intersect`
- SAM to BAM conversion, sorting and indexing is already done elsewhere; reuse and delete
- Diagnose and fix CI error

Optionally:
- Remove rule to convert GFF to BED for pri-miR counting; rewire to use GFF instead and update counting script accordingly

File: scripts/primir_quantification.py [hunk 1]
```diff
@@ -138,9 +138,9 @@ def main(args) -> None:
 
     with open(args.bedfile, 'r') as bedfile:
 
-        Fields = namedtuple('Fields',("feat_chr", "feat_start", "feat_end", 
-                                      "feat_id", "feat_score", "strand", "idk", 
-                                      "feat_type", "idkidk", "feat_attributes", 
+        Fields = namedtuple('Fields',("feat_chr", "NA", "feat_type",
+                                      "feat_start", "feat_end", "feat_score", 
+                                      "strand", "NaN", "feat_attributes",
```
Comment (uniqueg) [inline scripts/primir_quantification.py:143]: I think there's no need for `NA` and `NaN` column names. You can find the column/field meanings in the GFF format description:
https://www.ensembl.org/info/website/upload/gff3.html

**Review by uniqueg — CHANGES_REQUESTED** [review]
Please update the rule graph

**Review by uniqueg — APPROVED** [review]

**Pull request merged**